NDTensors/src/lib/Sectors/src/namedtuple_operations.jl

```diff
@@ -0,0 +1,16 @@
+
+@generated function nt_sort(nt::NamedTuple{N}) where {N}
+  return :(NamedTuple{$(Tuple(sort(collect(N))))}(nt))
+end
+
+@generated function nt_intersect(nt1::NamedTuple{N1}, nt2::NamedTuple{N2}) where {N1,N2}
+  return :(NamedTuple{$(Tuple(intersect(N1, N2)))}(merge(nt2, nt1)))
+end
+
+nt_union(ns1::NamedTuple, ns2::NamedTuple) = Base.merge(ns2, ns1)
+
+nt_setdiff(ns1::NamedTuple, ns2::NamedTuple) = Base.structdiff(ns1, ns2)
+
+function nt_symdiff(ns1::NamedTuple, ns2::NamedTuple)
+  return merge(Base.structdiff(ns1, ns2), Base.structdiff(ns2, ns1))
+end
```

Comment:
I think a better name for these would be `sort_keys`, `intersect_keys`, `union_keys`, etc. (then they would also generalize to `Dict` or `Dictionary`).

Reply:
Yes I had wanted to discuss these